forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -176,6 +179,259 @@ class DSRG_MRPT2 : public MASTER_DSRG {
+     * Xi:    constraint of the renormalized Fock matrix (F1) definition.
+     * Tau:   constraint of the two-body DSRG amplitude (T2) definition.
+     * Kappa: constraint of the renormalized ERIs (\tilde{V}) definition.
+     * Z:     OPDM, constraint of the CASSCF reference.
+     * W:     EWDM, constraint of the orthonormal overlap integral.
+     */
+    void set_multiplier();
+    /**
+     * Solve the Linear System Ax=b and yield Z using iterative methods.
+     */
+    void set_preconditioner(std::vector<double>&);
+    void gmres_solver(std::vector<double>&);
+    void solve_linear_iter();
+    void z_vector_contraction(std::vector<double> &, std::vector<double> &);
+    void pre_contract();
+    /**
+     * Solve the Linear System Ax=b and yield Z using direct methods.
+     */
+    void solve_z();
+    /**
+     * Initialize and solve the multiplier Sigma and Xi.
+     *
+     * Sigma: constraint of the one-body DSRG amplitude (T1) definition.
+     * Solved directly.
+     */
+    void set_sigma_xi();
+    /**
+     * Initialize and solve the multiplier Tau.
+     *
+     * Tau: constraint of the two-body DSRG amplitude (T2) definition.
+     * Solved directly.
+     */
+    void set_tau();
+    /**
+     * Initialize and solve the multiplier Kappa.
+     *
+     * Kappa: constraint of the renormalized ERIs (\tilde{V}) definition.
+     * Solved directly.
+     */
+    void set_kappa();
+    /**
+     * Initialize and solve the multiplier Z (OPDM).
+     *
+     * Z: OPDM, constraint of the CASSCF reference.
+     * The core-core, virtual-virtual blocks and diagonal entries of the active-active
+     * blocks are solved directly based on other multipliers. The rest need be solved
+     * Through iterative approaches. Currently, we use LAPACK as the solver.
+     */
+    void set_z();
+    /**
+     * Setting the b of the Linear System Ax=b.
+     * Parameters: preidx, block_dim
+     */
+    void set_b(int, std::map<string, int>, std::map<string, int>);
+    /**
+     * The diagonal core-core, virtual-virtual blocks 
+     * and the diagonal entries of the active-active block of the OPDM Z.
+     */
+    void set_z_diag();
+    /**
+     * Initialize and solve the multiplier W.
+     *
+     * W: EWDM, constraint of the orthonormal overlap integral.
+     * Solved directly after all other multipliers are solved.
+     */
+    void set_w();
+    // Set MO orbital partition info for solving the z-vector equation
+    void set_zvec_moinfo();
+
+    /// Size of determinants
+    size_t ndets;
+    /// List of core MOs (Correlated)
+    std::vector<size_t> core_mos;
+    /// List of active MOs (Correlated)
+    std::vector<size_t> actv_mos;
+    /// List of virtual MOs (Correlated)
+    std::vector<size_t> virt_mos;
+    /// List of core MOs (Absolute)
+
+    std::vector<size_t> core_all;
+    /// List of active MOs (Absolute)
+    std::vector<size_t> actv_all;
+    std::vector<size_t> virt_all;
```

Comment:
May I ask to rename these as `core_mos_all_`. Also add comment for virtual. A plausible comment would be `List of XXX MOs (including frozen orbitals)`.

Reply:
applied